tofu/geom/_core.py

```diff
@@ -4141,6 +4141,11 @@ def _complete_dX12(self, dgeom):
                         pinhole = dgeom['D'][:, 0] + k[0]*u[:, 0]
                         dgeom['pinhole'] = pinhole
 
+            if np.any(np.isnan(dgeom['D'])):
```

Comment:
This change to _core.py is just adding a safeguard (making it more robust with respect to ill-defined input from users)